pipeline/src/pipeline.rs

```diff
@@ -827,6 +828,12 @@ impl<T: FieldElement> Pipeline<T> {
         Ok(self.artifact.witness.as_ref().unwrap().clone())
     }
 
+    pub fn publics(&self) -> Result<Vec<(String, T)>, Vec<String>> {
```

Comment:
Wait but what does this do? Aren't publics always inputs?

Reply:
This retrieves the publics from the witness. It's useful for backends that need to pass the publics explicitly for the prover or a verifier

Reply:
The user passes the publics to powdr/pipeline, but for powdr they're just part of the witness table, with some copy constraints.

Reply:
Maybe this should be documented? I would not have expected a full witness computation just from asking for the publics.

Reply:
But thats not it, its just for when you already have the witness but just want to know what the publics are